src/ci_ex_states/excited_state_solver.cc

```diff
@@ -0,0 +1,1131 @@
+/*
+ * @BEGIN LICENSE
+ *
+ * Forte: an open-source plugin to Psi4 (https://github.com/psi4/psi4)
+ * that implements a variety of quantum chemistry methods for strongly
+ * correlated electrons.
+ *
+ * Copyright (c) 2012-2019 by its authors (see COPYING, COPYING.LESSER, AUTHORS).
+ *
+ * The copyrights for code used from other parties are included in
+ * the corresponding files.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ *
+ * @END LICENSE
+ */
+
+#include <fstream>
+#include <iomanip>
+#include <cmath>
+
+#include "psi4/libpsi4util/PsiOutStream.h"
+#include "psi4/physconst.h"
+#include "excited_state_solver.h"
+#include "sci/sci.h"
+#include "base_classes/mo_space_info.h"
+#include "sparse_ci/operator.h"
+#include "helpers/helpers.h"
+#include "helpers/printing.h"
+#include "ci_rdm/ci_rdms.h"
+namespace forte {
+ExcitedStateSolver::ExcitedStateSolver(StateInfo state, size_t nroot,
+                                       std::shared_ptr<MOSpaceInfo> mo_space_info,
+                                       std::shared_ptr<ActiveSpaceIntegrals> as_ints,
+                                       std::unique_ptr<SelectedCIMethod> sci)
+    : ActiveSpaceMethod(state, nroot, mo_space_info, as_ints), sci_(std::move(sci)) {
+    nact_ = mo_space_info_->size("ACTIVE");
+}
+
+void ExcitedStateSolver::set_options(std::shared_ptr<ForteOptions> options) {
+    // TODO: This shouldn't come from options
+    root_ = options->get_int("ROOT");
```

Comment:
Can't we get this from ActiveSpaceSolver?

Reply:
This one I copied from ACI, I have not think about this variable in detail.
https://github.com/evangelistalab/forte/blob/9ee177bff82d50b6e1e1de9487f3151598afcc9b/src/sci/aci.cc#L127-L128
I think you are right, but it seems that currently we do not have a good clue where we set this variable.